src/pod-security-policy/allow-privilege-escalation/src.cel

```diff
@@ -0,0 +1,26 @@
+variables:
+- name: containers
+  expression: 'has(object.spec.containers) ? object.spec.containers : []'
+- name: initContainers
+  expression: 'has(object.spec.initContainers) ? object.spec.initContainers : []'
+- name: ephemeralContainers
+  expression: 'has(object.spec.ephemeralContainers) ? object.spec.ephemeralContainers : []'
+- name: exemptImages
+  expression: |
+    !has(variables.params.exemptImages) ? [] :
+    (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+        // we can use the regex substitution because docker images do not allow "." as a valid character
+        variables.params.exemptImages.exists(exemption, container.image == exemption || (exemption.endsWith("*") && string(container.image).matches("^" + string(variables.params.exemptImages).replace("*", ".*") + "$")))
+    )
+- name: badContainers
+  expression: |
+    (variables.containers + variables.initContainers + variables.ephemeralContainers).filter(container,
+    !(container.image in variables.exemptImages) && (
+        !has(container.securityContext) ||
+        !has(container.securityContext.allowPrivilegeEscalation) ||
+        container.securityContext.allowPrivilegeEscalation != false
+    )
+    )
+validations:
+- expression: '(has(request.operation) && request.operation == "UPDATE") || size(variables.badContainers) == 0'
```

Comment:
The `UPDATE` guard is off... if request.operation is undefined, then the policy should be evaluated. Otherwise audit wont work.

Rego logic path:

* if not is_update(), continue processing violation
* is_update() is falsely if: `review.operation` is undefined or `review.operation == "UPDATE"`

Reply:
This works for audit since the || handles operation == ""

Reply:
Oops, you are correct.